yarnpkg/yarn
# Adding `docs` (alias: `home`) command

**Summary**

`npm home [package]` is one of my most used commands.  I really miss having that in `yarn`.  I've added `docs` as the main command (same as npm)- and I've also added the `yarn home` alias.  It works about the same as the npm command except I'm using [opn](https://www.npmjs.com/package/opn) instead of [opener](https://www.npmjs.com/package/opener), and I'm also using `--app` as the flag instead of `--browser`.  I can discuss more if you want.  Really hoping you can pull this in (as it's a command I really miss).  I'm not sure if you've had discussions about this in the past.  One other issue I had was that some of our dev dependencies are out of date.  I had issues w/ flow-types based on the version of jest we are using vs the flow type versions we are using.  Due to this fact, I had to use some `$FlowFixMe` comments.  Also, we are using an older version of jest, so I used `jest.clearAllMocks()`.  In newer versions of jest, I'd need to use `jest.resetAllMocks()`

**Test plan**

1. Run `yarn run test`
2. Confirm there are no lint or flow errors.
3. Confirm code coverage is 100% on all the `doc` releated files
4. Run `yarn build`
5. Do some smoke testing with the cli that was built:
  - `./bin/yarn docs --help`
  - `./bin/yarn docs`
  - `./bin/yarn home`
  - `./bin/yarn docs --app "Firefox" react flow redux`

**Commit Message**

- works similar to npm's `docs` command w/ the following caveats
  - using "opn" instead of "opener" library
  - using "--app" instead of "--browser"
- currently ignoring some jest flow-types. will update in another PR

**Quick Demo**

![yarn-docs](https://cloud.githubusercontent.com/assets/434470/21166659/e3f9f70c-c174-11e6-8ab2-17bd230aaf45.gif)
